main.go

```diff
@@ -73,18 +72,14 @@ func main() {
 	}
 	logger.SetLevel(logLevel)
 
-	// Set the preferred portal address.
-	database.Portal = os.Getenv("PORTAL_DOMAIN")
-	if database.Portal == "" {
-		log.Fatal("missing env var PORTAL_DOMAIN")
-	}
-	if !strings.HasPrefix(database.Portal, "http") {
-		database.Portal = "https://" + database.Portal
-	}
-	// Set the unique name of this server.
+	// Set the unique name of this server. If SERVER_DOMAIN is net present,
+	// which is possible in a single-server setup, the PORTAL_DOMAIN is used.
 	database.ServerDomain = os.Getenv("SERVER_DOMAIN")
 	if database.ServerDomain == "" {
-		log.Fatal("missing env var SERVER_DOMAIN")
+		database.ServerDomain = os.Getenv("PORTAL_DOMAIN")
```

Comment:
can we discuss this a bit... I don't feel good about this yet

Reply:
Imo this is a no-go. We rely on this code to block malicious content and we can't risk accidentally misconfiguring a server of our fleet without realising it because we forgot to set an env var. We've been known to forget setting vars so this is definitely going to happen at some point.

So this should not boot successfully if the variables aren't set correctly to unique values. 
If this means we have to introduce a `SERVER_UID` instead of the `SERVER_DOMAIN` then that's fine as well.
It might not be too bad after all cause that also gives users a way to set a more friendly unique name for a server than an URL.

Reply:
> Imo this is a no-go. We rely on this code to block malicious content and we can't risk accidentally misconfiguring a server of our fleet without realising it because we forgot to set an env var. We've been known to forget setting vars so this is definitely going to happen at some point.
> 
> So this should not boot successfully if the variables aren't set correctly to unique values. If this means we have to introduce a `SERVER_UID` instead of the `SERVER_DOMAIN` then that's fine as well. It might not be too bad after all cause that also gives users a way to set a more friendly unique name for a server than an URL.

there's nothing preventing a bad copy & paste of server uid either, it's as much of an env variable as server domain